Pirat/Services/Resource/ResourceUpdateService.cs

```diff
@@ -156,19 +157,135 @@ public Task ChangeConsumableAmount(string token, int consumableId, int newAmount
             return ChangeConsumableAmount(token, consumableId, newAmount, null);
         }
 
-        public Task ChangeConsumableAmount(string token, int consumableId, int newAmount, string reason)
+        public async Task ChangeConsumableAmount(string token, int consumableId, int newAmount, string reason)
         {
-            throw new NotImplementedException();
+            // Get consumable from database
+            var query = from o in _context.offer
+                join c in _context.consumable on o.id equals c.offer_id
+                where token.Equals(o.token)
+                      && c.id == consumableId
+                select c;
+            var foundConsumables = query.ToList();
+            if (foundConsumables.Count == 0)
+            {
+                throw new DataNotFoundException(Error.ErrorCodes.NOTFOUND_CONSUMABLE);
+            }
+            ConsumableEntity consumable = foundConsumables[0];
+            
+            // If amount has not changed: do nothing
+            if (consumable.amount == newAmount)
+            {
+                return;
+            }
+            
+            // If amount has increased: no reason required
+            if (consumable.amount < newAmount)
+            {
+                consumable.amount = newAmount;
+                consumable.Update(_context);
+                
+                // Add log
+                new Change()
+                {
+                    change_type = "INCREASE_AMOUNT",
+                    element_id = consumable.id,
+                    element_type = "consumable",
+                    reason = reason,
+                    timestamp = DateTime.Now
+                }.Insert(_context);
+                
+                return;
+            }
+            
+            // If amount has decreased: ensure that a reason is provided
+            if (reason == null || reason.Trim().Length == 0)
```

Comment:
We could give an empty string for reason always as default value. In the change class  we can set reason to an empty string as default value as well. I would do this in fact for all non-promitive types. This is later helpful when we use an equals or toString method for instances of reason.

Reply:
How does it work? What do I need to do?